### PR TITLE
tests: Ensure the IRQ is disabled before enabling it at run time

### LINKED ISSUE #21812: tests/arch/arm/arm_irq_advanced_features failed on reel_board.
**To Reproduce**
1. west build -b reel_board -p auto tests/arch/arm/arm_irq_advanced_features
2. west flash
3. See error

**Screenshots or console output**
*** Booting Zephyr OS build zephyr-v2.1.0-803-gf05cbb421ddf ***

Running test suite arm_irq_advanced_features
starting test - test_arm_dynamic_direct_interrupts
ASSERTION FAIL [!arch_irq_is_enabled(irq)] @ ZEPHYR_BASE/arch/common/sw_isr_common.c:19
IRQ 47 is enabled
E: r0/a1: 0x00000004 r1/a2: 0x00000013 r2/a3: 0x00000001
E: r3/a4: 0x00006664 r12/ip: 0x00000000 r14/lr: 0x00000db3
E: xpsr: 0x61000000
E: Faulting instruction address (r15/pc): 0x00005824
E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
E: Current thread: 0x2000004c (unknown)
E: Halting system

**Environment (please complete the following information):**
 - OS: fedora28
 - Toolchain: zephyr-sdk-0.10.3
 - Commit ID: f05cbb42

### PR DESCRIPTION
**Pull request by activebug202**
In the function irq_connect_dynamic, it will check if the irq is
disable. If the irq is enable ,it will assert.So before invoking
the function irq_connect_dynamic, it must disable the irq.

Signed-off-by: box zheng <box.zheng@intel.com>

Fixes #21812

### CODE REVIEW

**Review by LeiW000 — CHANGES_REQUESTED** [review]
First line of commit log needs to be changed. It should be a one-line summary of your change, rather than saying the issue id fixed.

I'm wondering why Ioannis did not see the issue when adding this test case. Which platform was used for testing by Ioannis? @ioannisg

**Comment by ioannisg** [comment]
> First line of commit log needs to be changed. It should be a one-line summary of your change, rather than saying the issue id fixed.
> 
> I'm wondering why Ioannis did not see the issue when adding this test case. Which platform was used for testing by Ioannis? @ioannisg

I did not use the reel board; instead I 've used standard nRF dev-kits to test the implementation :)

**Review by ioannisg — COMMENTED** [review]
Could you please add a more descriptive commit title, e.g. _tests: arch: arm_irq_advanced_features: ......_ ?

File: tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c [hunk 1]
```diff
@@ -35,6 +35,8 @@ void test_arm_dynamic_direct_interrupts(void)
 	 * in the ROM ISR table.
 	 */
 	ARM_IRQ_DIRECT_DYNAMIC_CONNECT(DIRECT_ISR_OFFSET, 0, 0, no_reschedule);
+	/* Disable the interrupt */
```
Comment (ioannisg) [inline tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c]: Maybe rephrase to, "Ensure the IRQ is disabled before enabling it at run time"

**Review by ioannisg — CHANGES_REQUESTED** [review]
Thanks for the bug fix, I 've tested that disabling the IRQ before connecting it dynamically works in this board. Please, implement the minor changes I suggested.

File: tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c (hunk 1 above)
Comment (activebug202) [reply tests/arch/arm/arm_irq_advanced_features/src/arm_dynamic_direct_interrupts.c]: done

**Review by ioannisg — APPROVED** [review]

**Review by LeiW000 — APPROVED** [review]

**Review by chen-png — APPROVED** [review]
it's ok to fix #21812

**Review by maksimmasalski — APPROVED** [review]
